devicetypes/smartthings/fidure-2nd-generation-thermostat.src/fidure-2nd-generation-thermostat.groovy

```diff
@@ -0,0 +1,810 @@
+				break
+				case "0008":
+				map.name = "heatingDemand"
+				map.value = Integer.parseInt("${descMap.value}", 16)
+				break
+				case "0010":
+				Integer encoding = Integer.parseInt("${descMap.encoding}",16)
+					if (encoding == 0x30) {
+					map.name ="fanMode"
+					map.value = getFanModeMap()[descMap.value]
+					}
+				break
+	  			case "0011":
+				 if (descMap.encoding ==  "29") {
+					map.name = "coolingSetpoint"
+	  				map.value = getTemperature(descMap.value)
+				} // as opposed to private attr 0x11 (descMap.encoding ==  "23") 
+	  			break
+	  			case "0012":
+	  				map.name = "heatingSetpoint"
+	  				map.value = getTemperature(descMap.value)
+	  			break
+				case "0015": 
+					device.updateDataValue("minheatSetpoint",  "${Integer.parseInt(descMap.value, 16)}")
+                break
+	            case "0016": 
+					device.updateDataValue("maxheatSetpoint",  "${Integer.parseInt(descMap.value, 16)}")
+                break
+		        case "0017": 
+					device.updateDataValue("mincoolSetpoint",  "${Integer.parseInt(descMap.value, 16)}")	
+                break
+			    case "0018": 
+					device.updateDataValue("maxcoolSetpoint",  "${Integer.parseInt(descMap.value, 16)}")	
+                break
+	  			case "001c":
+                    map.name = "thermostatMode"
+	  				map.value = getModeMap()[descMap.value]
+	  			break
+				case "001e":   
+			      	map.name = "runningMode"
+					map.value = getModeMap()[descMap.value]
+				break
+	            case "0023":   
+	            map.name = "setpointHold"
+	            map.value = getOnOffMap()[descMap.value]
+				sendEvent("name":"setpointHold", "value":getOnOffMap()[descMap.value])
+	            break
+	            case "0024":   
+	            map.name = "setpointHoldDuration"
+	            map.value = Integer.parseInt("${descMap.value}", 16)
+	            break
+	            case "0025":   
+				map.name = "prorgammingOperation"
+	            map.value = getOnOffMap()[descMap.value]
+				sendEvent("name":"prorgammingOperation", "value":getOnOffMap()[descMap.value])
+	  			break
+	            case "0029":
+	            map.name = "thermostatOperatingState"
+	            map.value = getThermostatOperatingState(descMap.value)
+	            break
+	      }
+
+		  return map
+}
+
+
+private Map parseCatchAllMessage(String description) {
+	Map resultMap = [:]
+    def cluster = zigbee.parse(description)
+    return resultMap
+}
+
+def setheatingSetpointAfterDelay(data){
+	sendZigbeeCmds(data.c, 100)
+}
+
+def setcoolingSetpointAfterDelay(data){
+	sendZigbeeCmds(data.c, 100)
+}
+
+def setSetpoint(Integer attrId, Double value){
+	Integer zigBeeTemp = getCelsiusValueX100(value)
+    
+	def cmds = zigbee.writeAttribute(0x0201, attrId , 0x29 , zigBeeTemp) 
+}
+
+def setHeatingSetpoint(Double degrees) {
+	
+	sendEvent(name: "heatingSetpoint", value: degrees)
+	setSetpoint(0x12, degrees)
+}
+
+def setCoolingSetpoint(Double degreesF) {
```

Comment:
degreesF ?

Reply:
fixed.  was infected from here: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/905096b3814980ac0b28291439a73d8e7c52e644/devicetypes/smartthings/tile-ux/tile-multiattribute-thermostat.src/tile-multiattribute-thermostat.groovy

there seems to be more of them throughout the repository.